google-cloud-tools-plugin/testSrc/com/google/cloud/tools/intellij/apis/CloudApiManagementConfirmationDialogTest.java

```diff
@@ -0,0 +1,138 @@
+import com.intellij.testFramework.fixtures.IdeaProjectTestFixture;
+import java.util.Set;
+import org.junit.Rule;
+import org.junit.Test;
+
+/** Tests for {@link CloudApiManagementConfirmationDialog}. */
+public class CloudApiManagementConfirmationDialogTest {
+
+  @Rule public final CloudToolsRule cloudToolsRule = new CloudToolsRule(this);
+  @TestFixture private IdeaProjectTestFixture testFixture;
+  @TestModule Module module;
+
+  private static final CloudProject cloudProject = CloudProject.create("name", "id", "user");
+
+  @Test
+  public void enablementUiState_whenAllApisAreEnabled() {
+    Set<CloudLibrary> libraries = ImmutableSet.of(TestCloudLibrary.createEmpty().toCloudLibrary());
+
+    ApplicationManager.getApplication()
+        .invokeAndWait(
+            () -> {
+              CloudApiManagementConfirmationDialog dialog =
+                  new CloudApiManagementConfirmationDialog(
+                      module, cloudProject, libraries, ImmutableSet.of(), ImmutableSet.of());
+
+              assertThat(dialog.getApisToEnablePanel().isVisible()).isTrue();
+              assertThat(dialog.getApisNotSelectedToEnablePanel().isVisible()).isFalse();
+            });
+  }
+
+  @Test
+  public void enablementUiState_whenSomeApisAreEnabled() {
+    Set<CloudLibrary> librariesToEnable =
+        ImmutableSet.of(TestCloudLibrary.createEmpty().toCloudLibrary());
+    Set<CloudLibrary> librariesNotToEnable =
+        ImmutableSet.of(TestCloudLibrary.createEmpty().toCloudLibrary());
+
+    ApplicationManager.getApplication()
+        .invokeAndWait(
+            () -> {
+              CloudApiManagementConfirmationDialog dialog =
+                  new CloudApiManagementConfirmationDialog(
+                      module,
+                      cloudProject,
+                      librariesToEnable,
+                      librariesNotToEnable,
+                      ImmutableSet.of());
+
+              assertThat(dialog.getApisToEnablePanel().isVisible()).isTrue();
+              assertThat(dialog.getApisNotSelectedToEnablePanel().isVisible()).isTrue();
+            });
+  }
+
+  @Test
+  public void enablementUiState_whenNoApisAreEnabled() {
+    Set<CloudLibrary> librariesNotToEnable =
+        ImmutableSet.of(TestCloudLibrary.createEmpty().toCloudLibrary());
+
+    ApplicationManager.getApplication()
+        .invokeAndWait(
+            () -> {
+              CloudApiManagementConfirmationDialog dialog =
+                  new CloudApiManagementConfirmationDialog(
+                      module,
+                      cloudProject,
+                      ImmutableSet.of(),
+                      librariesNotToEnable,
+                      ImmutableSet.of());
+
+              assertThat(dialog.getApisToEnablePanel().isVisible()).isFalse();
+              assertThat(dialog.getApisNotSelectedToEnablePanel().isVisible()).isTrue();
+            });
+  }
+
+  @Test
+  public void serviceAccountUi_whenSomeRolesExist_isVisible() {
```

Comment:
We could also add a test later which checks that role table properly populates and sets default state for roles, WDYT?

Reply:
Good idea

Reply:
done